src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -1614,6 +1614,59 @@ class spell_gen_elune_candle : public SpellScript
     }
 };
 
+// 50051 - Ethereal Pet Aura
+enum EtherealPet
+{
+    SPELL_STEAL_ESSENCE_VISUAL          = 50101,
+    SPELL_CREATE_TOKEN                  = 50063
+};
+
+class spell_ethereal_pet_aura : public AuraScript
+{
+    PrepareAuraScript(spell_ethereal_pet_aura);
+
+    void OnProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
```

Comment:
I read in somewhere, that you can just get credit if diff of levels between you and killed monster is <= 9

Reply:
Can you explain what do you mean?

Reply:
The same level diff as required to get XP?

Reply:
```cpp
int32 levelDiff = GetTarget()->getLevel() - eventInfo.GetProcTarget()->getLevel();
if (levelDiff <= 9 && levelDiff >= -9)
```

Reply:
uint32 levelDiff = std::abs(GetTarget()->getLevel() - eventInfo.GetProcTarget()->getLevel());
if (levelDiff <= 9)

Reply:
^ yeah, better